### DISCUSSION TITLE
How long can a Serverless function keep running

### DISCUSSION THREAD

**Discussion by ahmednabik** [vercel/community#5013] [Feature Question] [answered]
### Question

I recently deployed my first Express backend api on vercel (I am new to development). One of the API handlers posts on social media but puts a delay between two consecutive posts using setTimeout of 5 minutes. It works great on my localhost dev environment but will the same work with serverless function? I checked the vercel docs and found something they refer to as [Max Duration](https://vercel.com/docs/functions/serverless-functions/runtimes#max-duration) but I am not sure if this is the same thing a function keeps running  or not.

**Comment by swarnava** [comment, marked as answer]
Hi,

By default Serverless function can run upto 10 seconds on Hobby, 15 seconds on Pro, and 900 seconds on Enterprise plan. If you are a Pro customer, you can increase Serverless function timeout upto 300 seconds using `maxDuration` [config](https://vercel.com/docs/functions/configuring-functions/duration)
